### PR TITLE
Adds new group listing block

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

This PR adds a new block type: "Group listing block". Rather than re-use the listing block and add more fields I thought it best to create a new one along with a 'header/title' you can see in the [designs](https://www.figma.com/file/goD9bRqPoedQAxi8OsInkR/Data-Futures-Lab%3A-Designs?node-id=644%3A8116).

image

Link to sample test page: http://localhost:8000/en/campaigns/initial-test-bannered-campaign-with-fixed-title/ > Add new group listing block

Related PRs/issues: #9403  #9404 

# Checklist

**Tests**
- ~~[ ] Is the code I'm adding covered by tests?~~

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~
- [ ] ~~Did I squash my migration?~~
- [ ] ~~Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- [ ] ~~Is my code documented?~~
- [ ] ~~Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

**Comment by stevedya** [comment]
hey @kevinhowbrook i've made some changes to this one now: 

- Changed to tailwind's grid classes for braekpoints
- Used gap for spacing between grid columns
- Updated tailwind to be able to take advantage of arbitrary variants for styling paragraphs inside rich text block
- Made the entire card a link, but kept the aria-label as the card title
- Adjusted image size (just doubled it) so that it can be clearer when you get near tablet breakpoints
- Adjusted font size of the rich text elements to match closer to figma
- Used tailwind's space-y class to space out the rows of cards. [Reference](https://tailwindcss.com/docs/space#add-vertical-space-between-children)

**Screenshot**
Screen Shot 2022-10-25 at 6 23 54 PM

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html:38]: What do you think about using a `ul` here since this is semantically a listing?

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html:27]: What do you think about actually using an `h3` here since we also add the previous `h2`?

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html:24]: Can yo add some context why we are using `span`s here? It looks like we want them to act like block rather than inline elements.

**Review by tbrlpld — COMMENTED** [review]
@kevinhowbrook @stevedya Looks good overall. I have left a few small questions.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (stevedya) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html]: Sure! I've made this change now

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (stevedya) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html]: I actually juggled with this idea as well so i'm happy to update that. I've made the change now

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html
Comment (stevedya) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/group_listing_block.html]: WIth this being a h3 now we no longer need the block class so i've removed it now 😄  good catch

**Comment by stevedya** [comment]
@tbrlpld Great feedback. I've adjusted the code to resolve it now. It's ready for another look now

**Review by tbrlpld — APPROVED** [review]
Nice one.

**Comment by tbrlpld** [comment]
Thanks @stevedya. Nice work. I'll merge this after #9519 is merged.

**Pull request merged**
